From patchwork Mon Nov  5 14:04:55 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: channel_curl: fix compilation for curl version >= 7.62.0
X-Patchwork-Submitter: Christian Storm <christian.storm@siemens.com>
X-Patchwork-Id: 993079
Message-Id: <20181105140455.17505-1-christian.storm@siemens.com>
To: swupdate@googlegroups.com
Cc: Christian Storm <christian.storm@siemens.com>
Date: Mon,  5 Nov 2018 15:04:55 +0100
From: Christian Storm <christian.storm@siemens.com>
List-Id: <swupdate.googlegroups.com>

Fix gcc error: duplicate case value since CURLE_SSL_CACERT
is defined to CURLE_PEER_FAILED_VERIFICATION as of curl
version 7.62.0.

Signed-off-by: Christian Storm <christian.storm@siemens.com>
---
 corelib/channel_curl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/corelib/channel_curl.c b/corelib/channel_curl.c
index a9f80a2..e81f631 100644
--- a/corelib/channel_curl.c
+++ b/corelib/channel_curl.c
@@ -324,7 +324,9 @@ channel_op_res_t channel_map_curl_error(CURLcode res)
 	case CURLE_INTERFACE_FAILED:
 	case CURLE_SSL_CONNECT_ERROR:
 	case CURLE_PEER_FAILED_VERIFICATION:
+#if LIBCURL_VERSION_NUM < 0x073E00
 	case CURLE_SSL_CACERT:
+#endif
 	case CURLE_USE_SSL_FAILED:
 		return CHANNEL_ENONET;
 	case CURLE_OPERATION_TIMEDOUT:
